File: ch8/py/f-edit-state.py
from langgraph.graph import StateGraph
from langgraph.checkpoint.memory import MemorySaver


def main():
    # Create a simple graph
    builder = StateGraph()
    # Add nodes and edges as needed
    graph = builder.compile(checkpointer=MemorySaver())

    config = {"configurable": {"thread_id": "1"}}

    state = graph.get_state(config)
    print("Current state:", state)

    # something you want to add or replace
    update = {}

    graph.update_state(config, update)
    print("State updated")


if __name__ == "__main__":
    main()

File: ch8/js/f-edit-state.js
import { MemorySaver } from "@langchain/langgraph";
import { StateGraph } from "@langchain/langgraph";
import { Annotation } from "@langchain/langgraph";
import * as dotenv from "dotenv/config";
dotenv;

const GraphState = Annotation.Root({
  messages: Annotation(),
});

// Assuming graph is already created and configured
const graph = new StateGraph(GraphState).compile({
  checkpointer: new MemorySaver(),
});

const config = { configurable: { thread_id: "1" } };

const state = await graph.getState(config);
console.log("Current state:", state);

// something you want to add or replace
const update = {};

await graph.updateState(config, update);
console.log("State updated", state);
